pkg/app/two_pass_renderer.go

```diff
@@ -99,16 +103,22 @@ func (r *desiredStateLoader) twoPassRenderTemplateToYaml(inherited, overrode *en
 	if runtime.V1Mode {
 		var err error
 
-		finalEnv = initEnv
+		finalEnv, err = initEnv.Merge(overrode)
+		if err != nil {
+			return nil, err
+		}
 
 		vals, err = finalEnv.GetMergedValues()
 		if err != nil {
 			return nil, err
 		}
 	} else {
 		r.logger.Debugf("first-pass uses: %v", initEnv)
-
-		renderedEnv, prestate := r.renderPrestate(initEnv, baseDir, filename, content)
+		firstPassEnv, err := initEnv.Merge(nil)
+		if err != nil {
+			return nil, err
+		}
+		renderedEnv, prestate := r.renderPrestate(firstPassEnv, overrode, baseDir, filename, content)
```

Comment:
Note: We now apply overrides within renderPrestaste, instead of doing it beforehand.